recipes/libgd/all/conanfile.py

```diff
@@ -0,0 +1,76 @@
+import os
+from conans import ConanFile, tools, CMake
+
+
+class LibgdConan(ConanFile):
+    name = "libgd"
+    license = "https://github.com/libgd/libgd/blob/master/COPYING"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = ("GD is an open source code library for the dynamic "
+                   "creation of images by programmers.")
+    topics = ("images", "graphics")
+    settings = "os", "compiler", "build_type", "arch"
+    homepage = "https://libgd.github.io"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+    exports_sources = "CMakeLists.txt", "patches/**"
+    generators = "cmake"
+    requires = "zlib/1.2.11"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def _patch(self):
+        cmakelists = os.path.join(self._source_subfolder, "CMakeLists.txt")
+        tools.replace_in_file(cmakelists, "${CMAKE_SOURCE_DIR}", "${CMAKE_CURRENT_SOURCE_DIR}")
+        for patch in self.conan_data["patches"][self.version]:
+            tools.patch(**patch)
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        for name in ['libgd-', 'libgd-gd-']:
+            unpacked = name + self.version
+            if os.path.exists(unpacked):
+                os.rename(unpacked, self._source_subfolder)
+                break
+        else:
+            raise RuntimeError('Did not find the unpacked archive')
+
+    def _configure_cmake(self):
+        cmake = CMake(self)
+        cmake.definitions['BUILD_STATIC_LIBS'] = not self.options.shared
+        zlib_info = self.deps_cpp_info["zlib"]
+        cmake.definitions["ZLIB_LIBRARY"] = zlib_info.libs[0]
+        cmake.definitions["ZLIB_INCLUDE_DIR"] = zlib_info.include_paths[0]
+        cmake.configure(
+            source_folder=self.source_folder,
+            build_folder=self._build_subfolder)
+        return cmake
```

Comment:
```suggestion
        if self._cmake:
            return self._cmake
        self._cmake = CMake(self)
        self._cmake.definitions['BUILD_STATIC_LIBS'] = not self.options.shared
        zlib_info = self.deps_cpp_info["zlib"]
        self._cmake.definitions["ZLIB_LIBRARY"] = zlib_info.libs[0]
        self._cmake.definitions["ZLIB_INCLUDE_DIR"] = zlib_info.include_paths[0]
        self._cmake.configure(build_folder=self._build_subfolder)
        return self._cmake
```

* re-use the same cmake instance
* you can't use source_folder as _souce_subfolder, otherwise the cmake wrapper will be ignored.